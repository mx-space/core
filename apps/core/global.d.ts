import { Consola } from 'consola'
import { Document, PaginateModel } from 'mongoose'

import 'zx-cjs/globals'

import { ModelType } from '@typegoose/typegoose/lib/types'

declare global {
  export type KV<T = any> = Record<string, T>

  // @ts-ignore
  export type MongooseModel<T> = ModelType<T> & PaginateModel<T & Document>

  export const isDev: boolean

  export const consola: Consola

  export const cwd: string
}

export {}
