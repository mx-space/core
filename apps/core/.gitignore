schema.gql

app.config.js

webpack-dist

./admin
dist

tmp
