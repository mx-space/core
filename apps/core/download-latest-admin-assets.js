#!env node
const { appendFileSync } = require('fs')
const { join } = require('path')
const { fetch, $ } = require('zx-cjs')
const {
  dashboard: { repo, version },
} = require('./package.json')

const endpoint = `https://api.github.com/repos/${repo}/releases/tags/v${version}`
!(async () => {
  const json = await fetch(endpoint).then((res) => res.json())
  const downloadUrl = json.assets.find(
    (asset) => asset.name === 'release.zip',
  ).browser_download_url
  const buffer = await fetch(downloadUrl).then((res) => res.arrayBuffer())
  appendFileSync(join(process.cwd(), 'admin-release.zip'), Buffer.from(buffer))

  await $`ls -lh`

  await $`unzip admin-release.zip -d out`
  await $`mv out/dist out/admin`
  await $`rm -f admin-release.zip`
  // release.zip > dist > index.html
})()
