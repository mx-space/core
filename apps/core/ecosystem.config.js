const { cpus } = require('node:os')
const { execSync } = require('node:child_process')
const nodePath = execSync(`npm root --quiet -g`, { encoding: 'utf-8' }).split(
  '\n',
)[0]

const cpuLen = cpus().length
module.exports = {
  apps: [
    {
      name: 'mx-server',
      script: './index.js',
      autorestart: true,
      exec_mode: 'cluster',
      watch: false,
      instances: cpuLen,
      max_memory_restart: '520M',
      args: '--color --encrypt_enable',
      env: {
        NODE_ENV: 'production',
        NODE_PATH: nodePath,
        MX_ENCRYPT_KEY: process.env.MX_ENCRYPT_KEY,
        PORT: process.env.PORT,
        // NOTE: if OOM happens, try to use jemalloc
        // https://blog.csdn.net/qq_21567385/article/details/135322697
        // LD_PRELOAD: '/usr/lib/x86_64-linux-gnu/libjemalloc.so',
      },
    },
  ],
}
