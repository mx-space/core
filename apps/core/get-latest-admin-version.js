const { resolve } = require('path')
const axios = require('axios')
const { fs } = require('zx-cjs')
const {
  dashboard: { repo },
} = require('./package.json')
const Package = require('../../package.json')
const endpoint = `https://api.github.com/repos/${repo}/releases/latest`

const latestVersion = async () => {
  const res = await axios.get(endpoint)
  return res.data.tag_name.replace(/^v/, '')
}
async function main() {
  const version = await latestVersion()

  Package.dashboard.version = version

  await fs.writeFile(
    resolve(__dirname, './package.json'),
    JSON.stringify(Package, null, 2),
  )

  console.log('Updated version to', version)
}

main()
