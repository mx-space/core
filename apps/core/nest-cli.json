{
  "collection": "@nestjs/schematics",
  "sourceRoot": "src",
  "compilerOptions": {
    "plugins": []
  }
}
