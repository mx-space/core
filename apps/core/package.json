{
  "name": "@mx-space/core",
  "version": "4.6.2",
  "author": "Innei <https://innei.ren>",
  "private": true,
  "license": "AGPLv3",
  "dashboard": {
    "repo": "mx-space/mx-admin",
    "version": "3.33.4"
  },
  "repository": {
    "directory": "mx-space/core",
    "url": "https://github.com/mx-space/core"
  },
  "homepage": "https://github.com/mx-space/core#readme",
  "issues": "https://github.com/mx-space/core/issues",
  "scripts": {
    "prebuild": "rimraf dist",
    "build": "nest build",
    "build:webpack": "nest build --webpack --webpackPath ./configs/webpack.config.js -c ./configs/nest-cli.webpack.json",
    "dev": "npm run start",
    "repl": "npm run start -- --entryFile repl",
    "bundle": "rimraf out && npm run build && cd dist/src && npx ncc build main.js -o ../../out -m -t && cd ../.. && chmod +x out/index.js",
    "format": "prettier --write \"src/**/*.ts\" \"test/**/*.ts\" \"packages/**/*.ts\"",
    "start": "cross-env NODE_ENV=development nest start -w --path tsconfig.json -- ",
    "start:debug": "cross-env NODE_ENV=development nest start --debug --watch",
    "start:cluster": "cross-env NODE_ENV=development nest start --watch -- --cluster --workers 2",
    "start:prod": "cross-env NODE_ENV=production node dist/src/main",
    "lint": "eslint \"{src,apps,libs,test,packages}/**/*.ts\" --fix",
    "prod": "cross-env NODE_ENV=production pm2-runtime start ecosystem.config.js",
    "prod:pm2": "cross-env NODE_ENV=production pm2 restart ecosystem.config.js",
    "prod:stop": "pm2 stop ecosystem.config.js",
    "prod:debug": "cross-env NODE_ENV=production nest start --debug --watch",
    "test": "NODE_ENV=development vitest",
    "test:watch": "NODE_ENV=development vitest -w",
    "docs": "npx @compodoc/compodoc -p tsconfig.json -s -d docs",
    "publish": "bump"
  },
  "bump": {
    "before": [
      "git pull --rebase",
      "pnpm i",
      "node get-latest-admin-version.js"
    ],
    "after": [
      "sh assets-push.sh"
    ],
    "changelog": true
  },
  "dependencies": {
    "@algolia/client-search": "^4.20.0",
    "@babel/core": "7.23.5",
    "@babel/plugin-transform-modules-commonjs": "7.23.3",
    "@babel/plugin-transform-typescript": "7.23.5",
    "@babel/types": "^7.23.3",
    "@fastify/cookie": "9.2.0",
    "@fastify/multipart": "8.0.0",
    "@fastify/static": "6.12.0",
    "@innei/next-async": "0.3.0",
    "@nestjs/cache-manager": "2.1.1",
    "@nestjs/common": "10.2.10",
    "@nestjs/core": "10.2.10",
    "@nestjs/event-emitter": "2.0.3",
    "@nestjs/mapped-types": "^2.0.4",
    "@nestjs/platform-fastify": "10.2.10",
    "@nestjs/platform-socket.io": "10.2.10",
    "@nestjs/schedule": "4.0.0",
    "@nestjs/throttler": "5.0.1",
    "@nestjs/websockets": "10.2.10",
    "@socket.io/redis-adapter": "7.2.0",
    "@socket.io/redis-emitter": "4.1.1",
    "@typegoose/auto-increment": "4.0.0",
    "@typegoose/typegoose": "12.0.0",
    "@types/jsonwebtoken": "9.0.5",
    "algoliasearch": "4.20.0",
    "axios": "^1.6.2",
    "axios-retry": "4.0.0",
    "bcrypt": "5.1.1",
    "cache-manager": "5.3.1",
    "cache-manager-ioredis-yet": "1.2.2",
    "class-transformer": "0.5.1",
    "class-validator": "0.13.2",
    "class-validator-jsonschema": "npm:@innei/class-validator-jsonschema@3.1.2",
    "commander": "11.1.0",
    "dayjs": "1.11.10",
    "ejs": "3.1.9",
    "form-data": "4.0.0",
    "get-image-colors": "4.0.1",
    "image-size": "1.0.2",
    "inquirer": "^9.2.12",
    "isbot": "3.7.1",
    "js-yaml": "^4.1.0",
    "json5": "2.2.3",
    "jsonwebtoken": "9.0.2",
    "jszip": "3.10.1",
    "linkedom": "0.16.4",
    "lodash": "^4.17.21",
    "lru-cache": "10.1.0",
    "marked": "10.0.0",
    "mime-types": "^2.1.35",
    "mkdirp": "^3.0.1",
    "mongoose": "8.0.1",
    "mongoose-aggregate-paginate-v2": "1.0.6",
    "mongoose-autopopulate": "1.1.0",
    "mongoose-lean-getters": "1.1.0",
    "mongoose-lean-virtuals": "0.9.1",
    "mongoose-paginate-v2": "1.7.31",
    "nanoid": "3.3.4",
    "nestjs-pretty-logger": "0.2.0",
    "node-machine-id": "1.1.12",
    "node-pty": "1.0.0",
    "nodemailer": "6.9.7",
    "pluralize": "^8.0.0",
    "qs": "6.11.2",
    "reflect-metadata": "0.1.13",
    "remove-markdown": "0.5.0",
    "rxjs": "7.8.1",
    "semver": "7.5.4",
    "slugify": "1.6.6",
    "snakecase-keys": "5.5.0",
    "ua-parser-js": "1.0.37",
    "vm2": "3.9.19",
    "wildcard-match": "5.1.2",
    "xss": "1.0.14",
    "zx-cjs": "7.0.7-0"
  },
  "devDependencies": {
    "@nestjs/cli": "10.2.1",
    "@nestjs/schematics": "10.0.3",
    "@nestjs/testing": "10.2.10",
    "@swc/core": "1.3.99",
    "@types/babel__core": "7.20.5",
    "@types/bcrypt": "5.0.2",
    "@types/cache-manager": "4.0.6",
    "@types/ejs": "3.1.5",
    "@types/get-image-colors": "4.0.5",
    "@types/js-yaml": "4.0.9",
    "@types/lodash": "4.14.202",
    "@types/mime-types": "2.1.4",
    "@types/mongoose-aggregate-paginate-v2": "1.0.12",
    "@types/node": "20.10.1",
    "@types/nodemailer": "6.4.14",
    "@types/qs": "6.9.10",
    "@types/remove-markdown": "0.3.4",
    "@types/semver": "7.5.6",
    "@types/ua-parser-js": "0.7.39",
    "@types/validator": "13.11.7",
    "@vercel/ncc": "0.38.1",
    "cron": "^3.1.6",
    "husky": "8.0.3",
    "ioredis": "5.3.2",
    "socket.io": "^4.7.2",
    "unplugin-swc": "1.4.3",
    "vite": "5.0.4",
    "vite-tsconfig-paths": "4.2.1",
    "vitest": "0.34.6",
    "zx": "7.2.3"
  },
  "optionalDependencies": {
    "mongodb-memory-server": "^9.1.1",
    "redis-memory-server": "^0.9.0"
  }
}