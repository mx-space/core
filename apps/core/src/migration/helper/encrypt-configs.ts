import { plainToInstance } from 'class-transformer'

import 'reflect-metadata'

import { ENCRYPT } from '~/app.config'
import { register } from '~/global/index.global'
import { generateDefaultConfig } from '~/modules/configs/configs.default'
import * as optionDtos from '~/modules/configs/configs.dto'
import { encryptObject } from '~/modules/configs/configs.encrypt.util'
import { IConfig, IConfigKeys } from '~/modules/configs/configs.interface'
import { getDatabaseConnection } from '~/utils/database.util'

console.log(ENCRYPT)

const allOptionKeys: Set<IConfigKeys> = new Set()
Object.entries(optionDtos).reduce((obj, [key, value]) => {
  const optionKey = (key.charAt(0).toLowerCase() +
    key.slice(1).replace(/Dto$/, '')) as IConfigKeys
  allOptionKeys.add(optionKey)
  return {
    ...obj,
    [`${optionKey}`]: value,
  }
}, {})

async function main() {
  await register()
  const connection = await getDatabaseConnection()
  const db = connection.db
  const configs: any[] = []
  await db
    .collection('options')
    .find()
    .forEach((current) => {
      configs.push(current)
    })

  const mergedConfig = generateDefaultConfig()
  configs.forEach((field) => {
    const name = field.name as keyof IConfig

    if (!allOptionKeys.has(name)) {
      return
    }

    const value = field.value
    mergedConfig[name] = { ...mergedConfig[name], ...value }
  })

  const encrypted = encryptObject(plainToInstance(IConfig as any, mergedConfig))

  for await (const [key, value] of Object.entries(encrypted)) {
    configs[key] = value
    await db.collection('options').updateOne(
      {
        name: key,
      },
      {
        $set: {
          value,
        },
      },
    )
  }

  await connection.close()
  process.exit(0)
}

main()
