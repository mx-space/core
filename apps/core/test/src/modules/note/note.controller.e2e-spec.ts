import { createE2EApp } from 'test/helper/create-e2e-app'
import { authPassHeader } from 'test/mock/guard/auth.guard'
import { MockingCountingInterceptor } from 'test/mock/interceptors/counting.interceptor'
import { authProvider } from 'test/mock/modules/auth.mock'
import { commentProvider } from 'test/mock/modules/comment.mock'
import { configProvider } from 'test/mock/modules/config.mock'
import { gatewayProviders } from 'test/mock/modules/gateway.mock'
import { countingServiceProvider } from 'test/mock/processors/counting.mock'
import { eventEmitterProvider } from 'test/mock/processors/event.mock'

import { APP_INTERCEPTOR } from '@nestjs/core'

import { OptionModel } from '~/modules/configs/configs.model'
import { NoteController } from '~/modules/note/note.controller'
import { NoteModel } from '~/modules/note/note.model'
import { NoteService } from '~/modules/note/note.service'
import { UserModel } from '~/modules/user/user.model'
import { UserService } from '~/modules/user/user.service'
import { HttpService } from '~/processors/helper/helper.http.service'
import { ImageService } from '~/processors/helper/helper.image.service'
import { TextMacroService } from '~/processors/helper/helper.macro.service'

import MockDbData from './note.e2e-mock.db'

describe('NoteController (e2e)', () => {
  let model: MongooseModel<NoteModel>
  const proxy = createE2EApp({
    controllers: [NoteController],
    providers: [
      NoteService,
      ImageService,

      {
        provide: APP_INTERCEPTOR,
        useClass: MockingCountingInterceptor,
      },

      commentProvider,

      {
        provide: TextMacroService,
        useValue: {
          async replaceTextMacro(text) {
            return text
          },
        },
      },
      HttpService,
      configProvider,

      UserService,
      ...eventEmitterProvider,
      ...gatewayProviders,
      authProvider,

      countingServiceProvider,
    ],
    imports: [],
    models: [NoteModel, OptionModel, UserModel],
    async pourData(modelMap) {
      // @ts-ignore
      const { model: _model } = modelMap.get(NoteModel) as {
        model: MongooseModel<NoteModel>
      }
      model = _model
      for await (const data of MockDbData) {
        await _model.create(data)
      }
    },
  })

  afterAll(async () => {
    await model.deleteMany({})
  })

  test('GET /notes', async () => {
    const { app } = proxy
    const res = await app.inject({
      method: 'GET',
      url: '/notes',
    })
    const data = res.json()
    expect(res.statusCode).toBe(200)

    data.data.forEach((d) => {
      delete d.id
      delete d._id
    })
    expect(data).toMatchSnapshot()
  })

  const createdNoteData: Partial<NoteModel> = {
    title: 'Note 2',
    text: 'Content 2',

    allowComment: true,
    // use cutsom date
    created: new Date('2023-01-17T11:01:57.851Z'),
  }

  test('POST /notes', async () => {
    const { app } = proxy
    const res = await app.inject({
      method: 'POST',
      url: '/notes',
      payload: createdNoteData,
      headers: {
        ...authPassHeader,
      },
    })

    const data = res.json()
    expect(res.statusCode).toBe(201)
    createdNoteData.id = data.id
    createdNoteData.nid = data.nid
    delete data.id
    expect(data).toMatchSnapshot()
  })

  test('PATCH /notes/:id', async () => {
    const { app } = proxy
    const res = await app.inject({
      method: 'PATCH',
      url: `/notes/${createdNoteData.id}`,
      payload: {
        title: 'Note 2 (updated)',
        text: `Content 2 (updated)`,
        mood: 'happy',
        weather: 'sunny',
      },
      headers: {
        ...authPassHeader,
      },
    })

    expect(res.statusCode).toBe(204)
  })

  test('Get patched note', async () => {
    const { app } = proxy
    const res = await app.inject({
      method: 'GET',
      url: `/notes/${createdNoteData.id}`,
      headers: {
        ...authPassHeader,
      },
    })

    expect(res.statusCode).toBe(200)
    const data = res.json()
    delete data.id
    expect(data).toMatchSnapshot()
  })

  test('GET /list/:id', async () => {
    const { app } = proxy
    const res = await app.inject({
      method: 'GET',
      url: `/notes/list/${createdNoteData.id}`,
    })

    expect(res.statusCode).toBe(200)
    const data = res.json()

    data.data.forEach((note) => {
      delete note.id
    })

    expect(data).toMatchSnapshot()
  })

  test('GET /notes/nid/:nid', async () => {
    const { app } = proxy
    const res = await app.inject({
      method: 'GET',
      url: `/notes/nid/${createdNoteData.nid}`,
    })

    expect(res.statusCode).toBe(200)
    const data = res.json()
    delete data.id
    delete data.data.id
    if (data.prev) {
      delete data.prev.id
    }
    if (data.next) {
      delete data.next.id
    }

    expect(data).toMatchSnapshot()

    expect(countingServiceProvider.useValue.updateReadCount).toBeCalled()
  })

  test('DEL /notes/:id', async () => {
    const { app } = proxy
    const res = await app.inject({
      method: 'DELETE',
      url: `/notes/${createdNoteData.id}`,
      headers: {
        ...authPassHeader,
      },
    })

    expect(res.statusCode).toBe(204)
  })

  it('should got 404 when get deleted note', async () => {
    const { app } = proxy
    {
      const res = await app.inject({
        method: 'GET',
        url: `/notes/${createdNoteData.id}`,
        headers: {
          ...authPassHeader,
        },
      })

      expect(res.statusCode).toBe(404)
    }
    {
      const res = await app.inject({
        method: 'GET',
        url: `/notes/nid/${createdNoteData.nid}`,
        headers: {
          ...authPassHeader,
        },
      })

      expect(res.statusCode).toBe(404)
    }
  })

  test('GET /latest', async () => {
    const { app } = proxy
    const res = await app.inject({
      method: 'GET',
      url: '/notes/latest',
    })

    expect(res.statusCode).toBe(200)
    const data = res.json()
    delete data.data.id
    delete data.next.id
    expect(data).toMatchSnapshot()
  })

  let mockDataWithLocationNid = 0

  const createMockDataWithLocation = async () => {
    const note = await model.create({
      title: 'Note 3',
      text: 'Content 3',
      allowComment: true,
      coordinates: {
        latitude: 20,
        longitude: 20,
      },
      location: 'location',
    })
    mockDataWithLocationNid = note.nid
    return () => model.deleteOne({ _id: note._id })
  }

  test('GET /, should hide field when not login', async () => {
    const app = proxy.app

    await createMockDataWithLocation()
    const res = await app.inject({
      method: 'GET',
      url: '/notes',
    })

    const json = res.json()
    expect(json.data[0].coordinates).toBeUndefined()
    expect(json.data[0].location).toBeUndefined()
  })

  test('GET /, should show field when login', async () => {
    const app = proxy.app

    const res = await app.inject({
      method: 'GET',
      url: '/notes',
      query: {
        select: '+coordinates',
      },
      headers: {
        ...authPassHeader,
      },
    })

    const json = res.json()
    expect(json.data[0].coordinates).toBeDefined()
  })

  test('GET /nid/:nid, should hide field when not login', async () => {
    const app = proxy.app

    const res = await app.inject({
      method: 'GET',
      url: `/notes/nid/${mockDataWithLocationNid}`,
    })

    const json = res.json()
    expect(json.data.coordinates).toBeUndefined()
    expect(json.data.location).toBeUndefined()
  })

  let mockDataWithPasswordNid = 0

  const createMockDataWithPassword = async () => {
    const note = await model.create({
      title: 'Note 4',
      text: 'Content 3',
      allowComment: true,
      password: 'password',
    })
    mockDataWithPasswordNid = note.nid
    return () => model.deleteOne({ _id: note._id })
  }
  test('GET /nid/:nid, should ban if has password', async () => {
    const app = proxy.app

    await createMockDataWithPassword()
    const res = await app.inject({
      method: 'GET',
      url: `/notes/nid/${mockDataWithPasswordNid}`,
    })

    expect(res.statusCode).toBe(403)
  })

  test('GET /nid/:nid, should show if has password and pass', async () => {
    const app = proxy.app

    const res = await app.inject({
      method: 'GET',
      url: `/notes/nid/${mockDataWithPasswordNid}`,
      query: {
        password: 'password',
      },
    })

    expect(res.statusCode).toBe(200)
  })

  test('GET /nid/:nid, should show if has login', async () => {
    const app = proxy.app

    const res = await app.inject({
      method: 'GET',
      url: `/notes/nid/${mockDataWithPasswordNid}`,
      headers: {
        ...authPassHeader,
      },
    })

    expect(res.statusCode).toBe(200)
  })

  test('GET /like/:id', async () => {
    const app = proxy.app

    const res1 = await app.inject({
      method: 'GET',
      url: `/notes/like/${mockDataWithPasswordNid}`,
      headers: {
        ...authPassHeader,
      },
    })

    expect(res1.statusCode).toBe(204)

    const res2 = await app.inject({
      method: 'GET',
      url: `/notes/like/${mockDataWithPasswordNid}`,
      headers: {
        ...authPassHeader,
      },
    })

    expect(res2.statusCode).toBe(400)
  })
})
