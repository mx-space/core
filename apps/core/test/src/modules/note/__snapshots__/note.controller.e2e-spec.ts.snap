// Vitest Snapshot v1

exports[`NoteController (e2e) > GET /latest 1`] = `
{
  "data": {
    "allow_comment": true,
    "comments_index": 0,
    "count": {
      "like": 0,
      "read": 0,
    },
    "created": "2021-03-20T00:00:00.000Z",
    "hide": false,
    "images": [],
    "modified": null,
    "music": [],
    "nid": 20,
    "text": "Content 20",
    "title": "Note 20",
    "topic": null,
  },
  "next": {
    "nid": 19,
  },
}
`;

exports[`NoteController (e2e) > GET /list/:id 1`] = `
{
  "data": [
    {
      "created": "2023-01-17T11:01:57.851Z",
      "nid": 21,
      "title": "Note 2 (updated)",
    },
    {
      "created": "2021-03-20T00:00:00.000Z",
      "nid": 20,
      "title": "Note 20",
    },
    {
      "created": "2021-03-19T00:00:00.000Z",
      "nid": 19,
      "title": "Note 19",
    },
    {
      "created": "2021-03-18T00:00:00.000Z",
      "nid": 18,
      "title": "Note 18",
    },
    {
      "created": "2021-03-17T00:00:00.000Z",
      "nid": 17,
      "title": "Note 17",
    },
  ],
  "size": 5,
}
`;

exports[`NoteController (e2e) > GET /notes 1`] = `
{
  "data": [
    {
      "allow_comment": true,
      "comments_index": 0,
      "count": {
        "like": 0,
        "read": 0,
      },
      "created": "2021-03-20T00:00:00.000Z",
      "hide": false,
      "images": [],
      "meta": null,
      "modified": null,
      "music": [],
      "nid": 20,
      "text": "Content 20",
      "title": "Note 20",
      "topic": null,
    },
    {
      "allow_comment": true,
      "comments_index": 0,
      "count": {
        "like": 0,
        "read": 0,
      },
      "created": "2021-03-19T00:00:00.000Z",
      "hide": false,
      "images": [],
      "meta": null,
      "modified": null,
      "music": [],
      "nid": 19,
      "text": "Content 19",
      "title": "Note 19",
      "topic": null,
    },
    {
      "allow_comment": true,
      "comments_index": 0,
      "count": {
        "like": 0,
        "read": 0,
      },
      "created": "2021-03-18T00:00:00.000Z",
      "hide": false,
      "images": [],
      "meta": null,
      "modified": null,
      "music": [],
      "nid": 18,
      "text": "Content 18",
      "title": "Note 18",
      "topic": null,
    },
    {
      "allow_comment": true,
      "comments_index": 0,
      "count": {
        "like": 0,
        "read": 0,
      },
      "created": "2021-03-17T00:00:00.000Z",
      "hide": false,
      "images": [],
      "meta": null,
      "modified": null,
      "music": [],
      "nid": 17,
      "text": "Content 17",
      "title": "Note 17",
      "topic": null,
    },
    {
      "allow_comment": true,
      "comments_index": 0,
      "count": {
        "like": 0,
        "read": 0,
      },
      "created": "2021-03-16T00:00:00.000Z",
      "hide": false,
      "images": [],
      "meta": null,
      "modified": null,
      "music": [],
      "nid": 16,
      "text": "Content 16",
      "title": "Note 16",
      "topic": null,
    },
    {
      "allow_comment": true,
      "comments_index": 0,
      "count": {
        "like": 0,
        "read": 0,
      },
      "created": "2021-03-15T00:00:00.000Z",
      "hide": false,
      "images": [],
      "meta": null,
      "modified": null,
      "music": [],
      "nid": 15,
      "text": "Content 15",
      "title": "Note 15",
      "topic": null,
    },
    {
      "allow_comment": true,
      "comments_index": 0,
      "count": {
        "like": 0,
        "read": 0,
      },
      "created": "2021-03-14T00:00:00.000Z",
      "hide": false,
      "images": [],
      "meta": null,
      "modified": null,
      "music": [],
      "nid": 14,
      "text": "Content 14",
      "title": "Note 14",
      "topic": null,
    },
    {
      "allow_comment": true,
      "comments_index": 0,
      "count": {
        "like": 0,
        "read": 0,
      },
      "created": "2021-03-13T00:00:00.000Z",
      "hide": false,
      "images": [],
      "meta": null,
      "modified": null,
      "music": [],
      "nid": 13,
      "text": "Content 13",
      "title": "Note 13",
      "topic": null,
    },
    {
      "allow_comment": true,
      "comments_index": 0,
      "count": {
        "like": 0,
        "read": 0,
      },
      "created": "2021-03-12T00:00:00.000Z",
      "hide": false,
      "images": [],
      "meta": null,
      "modified": null,
      "music": [],
      "nid": 12,
      "text": "Content 12",
      "title": "Note 12",
      "topic": null,
    },
    {
      "allow_comment": true,
      "comments_index": 0,
      "count": {
        "like": 0,
        "read": 0,
      },
      "created": "2021-03-11T00:00:00.000Z",
      "hide": false,
      "images": [],
      "meta": null,
      "modified": null,
      "music": [],
      "nid": 11,
      "text": "Content 11",
      "title": "Note 11",
      "topic": null,
    },
  ],
  "pagination": {
    "current_page": 1,
    "has_next_page": true,
    "has_prev_page": false,
    "size": 10,
    "total": 20,
    "total_page": 2,
  },
}
`;

exports[`NoteController (e2e) > GET /notes/nid/:nid 1`] = `
{
  "data": {
    "allow_comment": true,
    "comments_index": 0,
    "count": {
      "like": 0,
      "read": 0,
    },
    "created": "2023-01-17T11:01:57.851Z",
    "hide": false,
    "images": [],
    "liked": true,
    "modified": null,
    "mood": "happy",
    "music": [],
    "nid": 21,
    "text": "Content 2 (updated)",
    "title": "Note 2 (updated)",
    "topic": null,
    "weather": "sunny",
  },
  "next": {
    "created": "2021-03-20T00:00:00.000Z",
    "modified": null,
    "nid": 20,
    "title": "Note 20",
  },
  "prev": null,
}
`;

exports[`NoteController (e2e) > Get patched note 1`] = `
{
  "allow_comment": true,
  "comments_index": 0,
  "count": {
    "like": 0,
    "read": 0,
  },
  "created": "2023-01-17T11:01:57.851Z",
  "hide": false,
  "images": [],
  "modified": null,
  "mood": "happy",
  "music": [],
  "nid": 21,
  "text": "Content 2 (updated)",
  "title": "Note 2 (updated)",
  "weather": "sunny",
}
`;

exports[`NoteController (e2e) > POST /notes 1`] = `
{
  "allow_comment": true,
  "comments_index": 0,
  "count": {
    "like": 0,
    "read": 0,
  },
  "created": "2023-01-17T11:01:57.851Z",
  "hide": false,
  "images": [],
  "meta": null,
  "modified": null,
  "music": [],
  "nid": 21,
  "text": "Content 2",
  "title": "Note 2",
  "topic": null,
}
`;
